pySDC/implementations/controller_classes/controller_nonMPI.py

```diff
@@ -91,7 +91,7 @@ def __init__(self, num_procs, controller_params, description):
         if self.params.use_HotRod and self.params.HotRod_tol == np.inf:
             self.logger.warning('Hot Rod needs a detection threshold, which is now set to infinity, such that a restart\
  is never triggered!')
-        self.error_estimator = ErrorEstimator_nonMPI(self)
+        self.error_estimator = ErrorEstimator_nonMPI().get_estimator(self)
```

Comment:
So, the rest of adaptivity and resilience is part of the controller.. why not this one, too? Or, better, how can we get the other ones out of the controller?

Reply:
Naming is indeed a very difficult task. I can also make a separate resilience module. Adaptivity as a control of the step size makes the most sense inside the controller, but as we discussed earlier, this perhaps makes it difficult to give a custom adaptivity function. I will leave that here for now and make a resilience module, and then we can shift adaptivity to a better position later. I agree with all other comments.

Reply:
So, I wonder if we need more building blocks which make up "a" controller.. this way we could have just a controller, an adaptive controller, a resilient controller, both, etc.. This would be like the opposite we have now (one to rule them all)...

Reply:
I would prefer individual modules that the controllers interface, like I am doing here with the error estimator. Such that, for instance, you do controller.resilience() and if you have no resilience specified, nothing happens. I think this makes it easier to mix and match different thinks and most importantly avoids copy pasting a whole bunch of code.
What we could do is something like the hooks, but for whatever, rather than just recording data. Then you can put that in the switcher and this way I think we can get about the same, but with inheritance rather than modules.
So pick you poison, because there is going to be some mess either way: Do you prefer modules or inheritance.
I started writing this comment absolutely wanting modules, but now I think maybe inheritance is the way to go. I am pretty torn honestly...